### PR TITLE
sign only assemblies that doesn't have dependency of non signed nugets

### PR DESCRIPTION
**Pull request by donandren**
## What does the pull request do?

remove signing from assemblies having dependency of non signed nugets

## What is the current behavior?

everything is signed and causing big troubles to be used (especially in net47) like:

- `Could not load file or assembly 'ReactiveUI, Version=11.5.0.0, Culture=neutral, PublicKeyToken=null' or one of its dependencies. A strongly-named assembly is required. (Exception from HRESULT: 0x80131044)`
valid for:
ReactiveUI, DynamicData, Splat, SharpGen.Runtime etc.
- not being able to compile projects using avalonia nugets dependant from non signed nugets

## What is the updated/expected behavior with this PR?

fix the issues for avalonia nugets like Avalonia.Native, Avalonia.ReactiveUI etc.

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Avaloniaui.net with user documentation

### CODE REVIEW

**Comment by kekekeks** [comment]
Signing should be opt-out, not opt-in. The entire problem is here because of strong religious beliefs of a certain ReactiveUI maintainer.

File: src/Avalonia.Desktop/Avalonia.Desktop.csproj [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<Project Sdk="Microsoft.NET.Sdk">
```
Comment (kekekeks) [inline src/Avalonia.Desktop/Avalonia.Desktop.csproj:1]: Not merging if Avalonia.Desktop is not signed, sorry

**Review by kekekeks — CHANGES_REQUESTED** [review]

**Comment by kekekeks** [comment]
I'll investigate merging SharpGen.Runtime later today.

**Comment by donandren** [comment]
@kekekeks added ilrepack to avalonia.native, so there is no unsigned dependency

File: src/Avalonia.Native/Avalonia.Native.csproj [hunk 2]
```diff
@@ -23,9 +25,24 @@
...
+      <WorkingDirectory>$(MSBuildThisFileDirectory)bin\$(Configuration)\$(TargetFramework)</WorkingDirectory>
+    </PropertyGroup>
+    <ItemGroup>
+      <InputAssemblies Include="SharpGen.Runtime.dll;SharpGen.Runtime.COM.dll;" />
+    </ItemGroup>
+    <Message Text="MERGING: @(InputAssemblies->'%(Filename)') into %(Target.FullPath)" Importance="High" />
+    <ILRepack OutputType="$(OutputType)" MainAssembly="$(AssemblyName).dll"
+              OutputAssembly="$(AssemblyName).dll" InputAssemblies="@(InputAssemblies)"
+              WorkingDirectory="$(WorkingDirectory)" 
+              KeyFile="$(AssemblyOriginatorKeyFile)" DelaySign="True" />
```
Comment (kekekeks) [inline src/Avalonia.Native/Avalonia.Native.csproj:46]: 1) Why DelaySign?
2) Are sharpgen types hidden afterwards?

Comment (donandren) [reply src/Avalonia.Native/Avalonia.Native.csproj:46]: merging sharpgen into native it self is working fine. With signing there is some issue for the build, i didn't had time to look in details and didn't spend time to investigate more

**Comment by kekekeks** [comment]
@donandren Since we should sign everything **except** for ReactiveUI, I think we should just exclude it from signing rather than explicitly enabling strong names on per-project basis

**Pull request closed without merging**